service/history/workflow/mutable_state_impl.go

```diff
@@ -6573,6 +6573,7 @@ func (ms *MutableStateImpl) applyUpdatesToStateMachineNodes(
 		internalNode.InitialVersionedTransition = nodeMutation.InitialVersionedTransition
 		internalNode.LastUpdateVersionedTransition = nodeMutation.LastUpdateVersionedTransition
 		internalNode.TransitionCount++
+		// TODO: Cache invalidation. This must be fixed before state based replication is enabled.
```

Comment:
Which cache?

Reply:
The node's cache, it keeps the previously deserialized version of the state in memory.